File: Frontend-POS/RBMS-POS-Client/src/app/core/api/models/dashboard-overview-response-model.ts
/* tslint:disable */
/* eslint-disable */
import { DashboardKpiModel } from '../models/dashboard-kpi-model';
import { KitchenBreakdownModel } from '../models/kitchen-breakdown-model';
import { PeriodRevenueTrendModel } from '../models/period-revenue-trend-model';
import { RevenueTrendModel } from '../models/revenue-trend-model';
export interface DashboardOverviewResponseModel {
  hasTwoPeriods?: boolean;
  kitchenBreakdown?: Array<KitchenBreakdownModel> | null;
  period1Label?: string | null;
  period2Label?: string | null;
  periodRevenueTrend?: Array<PeriodRevenueTrendModel> | null;
  previous?: DashboardKpiModel;
  revenueTrend?: Array<RevenueTrendModel> | null;
  selected?: DashboardKpiModel;
}

File: Backend-POS/POS.Main/POS.Main.Business.Admin/Models/Dashboard/DashboardOverviewResponseModel.cs
namespace POS.Main.Business.Admin.Models.Dashboard;

public class DashboardOverviewResponseModel
{
    public DashboardKpiModel Selected { get; set; } = new();
    public DashboardKpiModel Previous { get; set; } = new();
    public List<KitchenBreakdownModel> KitchenBreakdown { get; set; } = new();
    public List<RevenueTrendModel> RevenueTrend { get; set; } = new();
}

public class DashboardKpiModel
{
    public decimal TotalSales { get; set; }
    public int OrderCount { get; set; }
    public int GuestCount { get; set; }
    public decimal AveragePerOrder { get; set; }
}

public class KitchenBreakdownModel
{
    public int CategoryType { get; set; }
    public string CategoryName { get; set; } = string.Empty;
    public int ItemCount { get; set; }
    public decimal? Percentage { get; set; }
}

public class RevenueTrendModel
{
    public DateTime Date { get; set; }
    public decimal TotalSales { get; set; }
}
